megamek/src/megamek/common/MechSummaryCache.java

```diff
@@ -338,6 +361,38 @@ private void saveCache() throws Exception {
         wr.close();
     }
 
+    private void refreshCache(long lastCheck, boolean ignoreUnofficial) {
+        loadReport = new StringBuffer();
+        loadReport.append("Refreshing unit cache:\n");
+        Vector<MechSummary> units = new Vector<>();
+        Set<String> knownFiles = new HashSet<>();
+        // Loop through current contents and make sure the file is still there.
+        // Note which files are represented so we can skip them if they haven't changed
+        for (MechSummary ms : m_data) {
+            if (interrupted) {
+                done();
+                return;
+            }
+            File source = ms.getSourceFile();
+            if (source.exists()) {
+                units.add(ms);
+                if (null == ms.getEntryName()) {
+                    knownFiles.add(source.toString());
+                } else {
+                    knownFiles.add(ms.getEntryName());
+                }
```

Comment:
Ternary canditate

Reply:
Ternaries are good for assignments, but I find them clumsy as arguments.